server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -208,6 +208,45 @@ public static HighlightBuilder highlight() {
      */
     public SearchSourceBuilder() {}
 
+    /**
+     * Copy constructor.
+     */
+    public SearchSourceBuilder(SearchSourceBuilder source) {
+        this.subSearchSourceBuilders = source.subSearchSourceBuilders;
+        this.postQueryBuilder = source.postQueryBuilder;
+        this.knnSearch = source.knnSearch;
+        this.rankBuilder = source.rankBuilder;
+        this.from = source.from;
+        this.size = source.size;
+        this.explain = source.explain;
+        this.version = source.version;
+        this.seqNoAndPrimaryTerm = source.seqNoAndPrimaryTerm;
+        this.sorts = source.sorts;
+        this.trackScores = source.trackScores;
+        this.trackTotalHitsUpTo = source.trackTotalHitsUpTo;
+        this.searchAfterBuilder = source.searchAfterBuilder;
+        this.sliceBuilder = source.sliceBuilder;
+        this.minScore = source.minScore;
+        this.timeout = source.timeout;
+        this.terminateAfter = source.terminateAfter;
+        this.storedFieldsContext = source.storedFieldsContext;
+        this.docValueFields = source.docValueFields;
+        this.scriptFields = source.scriptFields;
+        this.fetchSourceContext = source.fetchSourceContext;
+        this.fetchFields = source.fetchFields;
+        this.aggregations = source.aggregations;
+        this.highlightBuilder = source.highlightBuilder;
+        this.suggestBuilder = source.suggestBuilder;
+        this.rescoreBuilders = source.rescoreBuilders;
+        this.indexBoosts = source.indexBoosts;
+        this.stats = source.stats;
+        this.extBuilders = source.extBuilders;
+        this.profile = source.profile;
+        this.collapse = source.collapse;
+        this.pointInTimeBuilder = source.pointInTimeBuilder;
+        this.runtimeMappings = source.runtimeMappings;
+    }
```

Comment:
Could you add a test for this new constructor to SearchSourceBuilderTests? Ideally, when we add a new member that we forget about in the copy constructor, some test fails.

Reply:
I've replaced the copy constructor by the existing `shallowCopy` method.

Note: that method is tested, but I don't think it's super solid. If you add a new field to `SearchSourceBuilder`, you're probably not adding it to `SearchSourceBuilderTests::createSearchSourceBuilder`, so you're not testing it. I'm not sure what's a good simple solution to this. I don't think it's an issue of this PR though.

Reply:
thanks, I forgot we had shallowCopy already.